hrjobcontract/CRM/Hrjobcontract/Upgrader.php

```diff
@@ -1279,13 +1279,9 @@ public function upgrade_1030() {
     try {
       CRM_Core_DAO::executeQuery('ALTER TABLE `civicrm_hrhours_location` ADD PRIMARY KEY (`id`)');
       CRM_Core_DAO::executeQuery('ALTER TABLE `civicrm_hrhours_location` CHANGE `id` `id` INT(10) UNSIGNED NOT NULL AUTO_INCREMENT');
-    } catch (Exception $e) {
-      $tableStructure = CRM_Core_DAO::executeQuery('DESC civicrm_hrhours_location');
-
-      while ($tableStructure->fetch()) {
-        if ($tableStructure->Field == 'id' && ($tableStructure->Key != 'PRI' || stripos($tableStructure->Extra, 'auto_increment') === false)) {
-          throw $e;
-        }
+    } catch (PEAR_Exception $e) {
+      if (stripos($e->getCause()->userinfo, 'nativecode=1068') === false) {
```

Comment:
This looks better, but the readability is still bad. No one knows what nativecode 1068 means. Assigning the `stripos` result to a variable with a descriptive name will make it easier for other devs to understand what you're trying to check here.

Also, I don't understand the reason for throwing a new exception with a new message. Is there any reason for that?